### DISCUSSION TITLE
Table element: configurable columns?

### DISCUSSION THREAD

**Discussion by sebm** [twilio-labs/paste#1074] [Ideas] [answered]
Product has requested configurable columns on a Table page, akin to this functionality in Mac OS X Finder:

![Screen Shot 2021-01-06 at 12 52 34](https://user-images.githubusercontent.com/188161/103818837-18306d80-501e-11eb-8d0e-1726fa50c0b1.png)

To my knowledge this isn't currently provided by Paste Tables. Although I could implement something myself in "userspace" on top of Paste, this seems to be something with broad enough benefit to be worth implementing for all Paste customers.

What do you think, is this something you can build?

**Comment by SiTaggart** [comment, marked as answer]
Hi @sebm, right now I'm not sure this is something we'd implement in the component. It requires knowing too much about the data it's displaying, and we really favor placing data handling responsibility on the consumer to afford you the most flexibility.

We might provide the controls needed to do something like this, the lego bricks. I've seen in other products a menu button placed above the table, used to edit a table configuration before (though we don't actually support checkbox menu items right now, it's coming).

You could also have a button above the table that launches a [popover](https://paste.twilio.design/components/popover) with a [checkbox group](https://paste.twilio.design/components/checkbox#checkbox-group) of table columns that you wish to display. Upon the user making the selection, you use that to filter your table data and use the table components to render the result.

Unfortunately it's very similar to providing functionality like table sorting, or virtualization. We're likely _not_ going to provide these features any time soon, only provide the UI controls that should be used to action them and how they should look, and leave the data handling up to the consuming team.

**Reply by corinne-nagel** [reply]
Hey, checking in on this, @sebm did Simon answer your question?

**Reply by andioneto** [reply]
@SiTaggart Hi there 👋 I'm picking this feature work up for our team. I did a bit of testing with the Popover component, but I noticed a few things that may be problematic:
1. The popover does not scroll to fit content
2. The popover fixes the position of the scroll, so if there is overflow in the current view, the view "sticks"

I'm wondering if instead I can use a modal with checkboxes since the multi-action menu is going to be released later?
I'm getting inspiration from this [article](https://medium.com/design-with-figma/the-ultimate-guide-to-designing-data-tables-7db29713a85a) looking at item 5.

**Reply by andioneto** [reply]
FYI just created this little POC in code sandbox, lmk what you think: https://j11zu.csb.app/

**Reply by SiTaggart** [reply]
Modal seems perfectly acceptable.

For the 2 issues with popover, are you able to create a reproduction in a codesandbox too? They sound like bugs, but it's a little hard to visualise without seeing the actual outcome.

**Reply by andioneto** [reply]
I don't any more, I can re-create it though in a separate codesandbox if that is useful though 😄 I just pulled one of the first examples from the docs, put it into the sandbox, and added lots of copy on new lines. LMK And thank you for the feedback!

**Reply by SiTaggart** [reply]
If you think it's a problem, such as a bug we should address, then a codesandbox would be super useful. Thanks @andipants12

**Reply by andioneto** [reply]
Sure thing, I'll spin something up today and leave it here. Thanks!
